src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/FileStorageService.cs

```diff
@@ -98,5 +95,60 @@ private void FileSkippedCallback(object sender, TransferEventArgs e)
         {
             _logger.LogInformation("Transfer skips. {0} -> {1}.", e.Source, e.Destination);
         }
+
+        private static async Task<bool> ShouldTransferCallbackAsync(object source, object destination)
+        {
+            if (source is CloudBlockBlob blob)
+            {
+                await blob.FetchAttributesAsync();
+                return !IsMetaDataFile(blob);
+            }
+
+            return true;
+        }
+
+        private static bool IsMetaDataFile(CloudBlockBlob blob)
+        {
+            // The meta data file contains a metadata attribute referencing it's corresponding data file
+            return blob.Metadata.ContainsKey(DataFileKey);
+        }
+
+        private static async void ZipAllFilesToBlob(IEnumerable<CloudBlockBlob> files, CloudBlobDirectory directory)
+        {
+            var cloudBlockBlob = CreateBlobForAllFilesZip(directory);
+            var memoryStream = new MemoryStream();
+
+            var zipOutputStream = new ZipOutputStream(memoryStream);
+            zipOutputStream.SetLevel(1);
+
+            foreach (var file in files)
+            {
+                PutNextZipEntry(zipOutputStream, file);
+            }
+
+            zipOutputStream.Finish();
+            await TransferManager.UploadAsync(memoryStream, cloudBlockBlob);
+
+            zipOutputStream.Close();
+        }
+
+        private static void PutNextZipEntry(ZipOutputStream zipOutputStream, CloudBlob cloudBlob)
+        {
+            var zipEntry = new ZipEntry(GetZipEntryName(cloudBlob));
+            zipOutputStream.PutNextEntry(zipEntry);
+            cloudBlob.DownloadToStream(zipOutputStream);
+        }
+
+        private static string GetZipEntryName(CloudBlob cloudBlob)
+        {
+            return cloudBlob.Uri.Segments.Last();
+        }
+
+        private static CloudBlockBlob CreateBlobForAllFilesZip(CloudBlobDirectory directory)
+        {
+            var blob = directory.GetBlockBlobReference("all.zip");
```

Comment:
this will work for now but i suggest we actually use the name of the publication and release.

Reply:
Thanks, have just updated it.